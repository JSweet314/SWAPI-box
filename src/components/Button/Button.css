.button {
  margin: 0 2rem;
  padding: 1rem 3rem;
  border-radius: 5px 5px;
  background-color: #4B545F;
  color: #E0E9FA;
  cursor: pointer;
}

.selected {
  background-color: #18212A;
}