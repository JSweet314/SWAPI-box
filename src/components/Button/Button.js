import React from 'react';
import './Button.css';
import PropTypes from 'prop-types';

const Button = ({btnText}) => {
  return (
    <button className="button">
      {btnText}
    </button>
  );
};

Button.propTypes = {
  btnText: PropTypes.string.isRequired
};

export default Button;