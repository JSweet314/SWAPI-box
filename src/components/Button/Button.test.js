import React from 'react';
import Button from './Button';
import {shallow} from 'enzyme';

describe('button', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Button 
        btnText="people"
        isSelected='' 
        /* eslint-disable no-undef */
        handleOnClick={jest.fn}
        /* eslint-enable no-undef */
      />
    );
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});