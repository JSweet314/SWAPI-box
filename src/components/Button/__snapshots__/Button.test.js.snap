// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    btnText="people"
    handleOnClick={[Function]}
    isSelected=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "people",
      "className": "button ",
      "name": "people",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "people",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "people",
        "className": "button ",
        "name": "people",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "people",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
